cub/warp/warp_exchange.cuh

```diff
@@ -230,24 +226,7 @@ public:
    *   Items from exchange, converting between <em>striped</em> and
    *   <em>blocked</em> arrangements. May be aliased to @p input_items.
    */
-  template <typename OutputT>
-  __device__ __forceinline__ void
-  BlockedToStriped(const InputT (&input_items)[ITEMS_PER_THREAD],
-                   OutputT (&output_items)[ITEMS_PER_THREAD])
-  {
-    for (int item = 0; item < ITEMS_PER_THREAD; item++)
-    {
-      const int idx = ITEMS_PER_THREAD * lane_id + item;
-      temp_storage.items_shared[idx] = input_items[item];
-    }
-    WARP_SYNC(member_mask);
-
-    for (int item = 0; item < ITEMS_PER_THREAD; item++)
-    {
-      const int idx = LOGICAL_WARP_THREADS * item + lane_id;
-      output_items[item] = temp_storage.items_shared[idx];
-    }
-  }
+  using InternalWarpExchange::BlockedToStriped;
```

Comment:
I'm afraid we'll have go the duplication route here. When people follow the API, they have to arrive at the function signature. Having them to go into the implementation details is not great. It'd also prevent some potential issues with mismatch of specializations signatures. Could you please implement it in the following form:

```cpp
  Alg(InputT (&items)[ITEMS_PER_THREAD],
      OffsetT (&ranks)[ITEMS_PER_THREAD])
  {
    InternalWarpExchange::Alg(items, ranks);
  }
```

You can omit addressing this comment in this PR if you don't have time. I can file an issue and address this later.